orderBook.js

```diff
@@ -1,5 +1,64 @@
+
+function matchingOrder(existingBook, incomingOrder) {
+    let matchingOrder = existingBook.filter((item) => {
+        return item.type !== incomingOrder.type &&
+            item.price === incomingOrder.price
+    })
+    return matchingOrder
+}
+
+function updatesMatchingOrder(existingBook, incomingOrder) {
+    let updatedBook = []
+    const matchingIndex = existingBook.findIndex((item) => {
+        return incomingOrder.type !== item.type &&
+            incomingOrder.quantity === item.quantity &&
+            incomingOrder.price === item.price
+    })
+    const partialMatchingIndex = existingBook.findIndex((item) => {
+        return incomingOrder.type !== item.type &&
+            incomingOrder.price === item.price
+    })
```

Comment:
The index is the important piece, I'd actually move these out to functions and use them and then you would even need the matchingOrder function